org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLHover.java

```diff
@@ -95,16 +98,20 @@ public Hover doHover(DOMDocument xmlDocument, Position position, XMLHoverSetting
 	private Hover getTagHover(HoverRequest hoverRequest, Range tagRange, boolean open) {
 		hoverRequest.setTagRange(tagRange);
 		hoverRequest.setOpen(open);
+		List<String> contentValues = new ArrayList<String>();
 		for (IHoverParticipant participant : extensionsRegistry.getHoverParticipants()) {
 			try {
-				Hover hover = participant.onTag(hoverRequest);
-				if (hover != null) {
-					return hover;
+				String contentValue = participant.onTag(hoverRequest);
+				if (contentValue != null) {
+					contentValues.add(contentValue);
 				}
 			} catch (Exception e) {
 				LOGGER.log(Level.SEVERE, "While performing IHoverParticipant#onTag", e);
 			}
 		}
+		if (!contentValues.isEmpty()) {

```

Comment:
share this code in private static method `createHover` to use it with other on* method.